libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -288,6 +297,27 @@ void AP_ICEngine::update(void)
         }
     }
 
+    // check against redline RPM
+    if (redline_rpm > 0 && rpm.get_rpm(rpm_instance-1, rpm_value)) {
+        // update the filtered RPM value
+        filtered_rpm_value =  _rpm_filter.apply(rpm_value, AP::scheduler().get_loop_period_s());
+        if (filtered_rpm_value > redline_rpm) {
+            // check if RPM is over the redline value
+            if (!redline.flag) {
+                gcs().send_text(MAV_SEVERITY_INFO, "Engine: Above redline RPM");
+            }
+            redline.flag = true;
+        } else if (filtered_rpm_value < redline_rpm - (redline_rpm * 0.1f)) {
+            if (redline.flag) {
+                gcs().send_text(MAV_SEVERITY_INFO, "Engine: Below redline RPM");
```

Comment:
I think in general we should avoid text messages when everything is working as it should be.  Users get tired of messages that they don't need to act on and they start ignoring all messages.

Reply:
Fixed.